website_product_document_download_counter/models/product_document.py

```
@@ -0,0 +1,36 @@
+    )
+
+    def increment_download_count(self):
+        """
+        Increment the download count for a specific product document.
+        Caller must ensure appropriate permissions (e.g. via sudo()).
+        """
+        self.ensure_one()
+        if not self.active:
+            raise UserError(_("Can't download an inactive document"))
+        if not self.shown_on_product_page:
+            raise UserError(_("Can't download a document not shown on product page"))
```

Flake8 (7.2.0)

[error] line too long (85 > 79 characters)

(E501)